rust/lance/src/dataset/fragment/write.rs

```diff
@@ -67,6 +69,68 @@ impl<'a> FragmentCreateBuilder<'a> {
         self.write_impl(stream, schema, id).await
     }
 
+    async fn write_v2_impl(
+        &self,
+        stream: SendableRecordBatchStream,
+        schema: Schema,
+        id: u64,
+    ) -> Result<Fragment> {
+        let params = self.write_params.map(Cow::Borrowed).unwrap_or_default();
+        let progress = params.progress.as_ref();
+
+        Self::validate_schema(&schema, stream.schema().as_ref())?;
+
+        let (object_store, base_path) = ObjectStore::from_uri(self.dataset_uri).await?;
+        let filename = format!("{}.lance", Uuid::new_v4());
+        let mut fragment = Fragment::new(id);
+        let full_path = base_path.child(DATA_DIR).child(filename.clone());
+        let obj_writer = object_store.create(&full_path).await?;
+        let mut writer = lance_file::v2::writer::FileWriter::try_new(
+            obj_writer,
+            full_path.to_string(),
+            schema,
+            FileWriterOptions::default(),
+        )?;
+
+        progress.begin(&fragment, writer.multipart_id()).await?;
+
+        let break_limit = (128 * 1024).min(params.max_rows_per_file);
+
+        let mut broken_stream = break_stream(stream, break_limit)
+            .map_ok(|batch| vec![batch])
+            .boxed();
+        while let Some(batched_chunk) = broken_stream.next().await {
+            let batch_chunk = batched_chunk?;
+            writer.write_batches(batch_chunk.iter()).await?;
+        }
```

Comment:
Seems a little odd here we have to wrap the batch into a Vec. Why not just:


```suggestion
        let mut broken_stream = break_stream(stream, break_limit);
        while let Some(batch) = broken_stream.next().await {
            writer.write_batches([batch_chunk?].iter()).await?;
        }
```